.github/workflows/pr-validate.yaml

```diff
@@ -34,22 +34,29 @@ jobs:
         with:
           github-token: ${{ secrets.GITHUB_TOKEN }}
           script: |
+            var ref;  
             const url = `${process.env.server_url}/${process.env.repo}/actions/runs/${process.env.run_id}`
-            const { data: pull } = await github.rest.pulls.get({
-              ...context.repo,
-              pull_number: process.env.number
-            });
-            console.log('retrieved pull request')
+            if (context.eventName == 'pull_request') {
+              ref = pull.head.sha;
+              const { data: pull } = await github.rest.pulls.get({
```

Comment:
Why do you need to get the pull request? It seems like you don't do anything with the data that gets returned?

Reply:
The ref on line 40 should have been set at the end of the if block after the GET request. Then the create check at the bottom of the script will use that ref.